backend/src/main/java/ca/mcgill/purposeful/service/IdeaService.java

```diff
@@ -472,4 +474,59 @@ public void removeIdeaById(String uuid) {
     // remove idea
     ideaRepository.deleteById(uuid);
   }
+
+  /**
+   * Retrieve all the ideas that a user expressed interest in.
+   *
+   * @param email EMail of the user to retrieve the ideas for
+   * @return List of ideas that the user expressed interest in
+   */
+  @Transactional
+  public List<Idea> getIdeasByCollaborationRequest(String email) {
+    // Input validation
+    String error = "";
+    if (email == null || email.trim().length() == 0) {
+      error += "Email cannot be left empty! ";
+    }
+    if (error.length() > 0) {
+      throw new GlobalException(HttpStatus.BAD_REQUEST, error);
+    }
+
+    // Check if the user we are trying to modify does indeed exist
+    RegularUser user = regularUserRepository.findRegularUserByAppUserEmail(email);
+    if (user == null) {
+      throw new GlobalException(HttpStatus.BAD_REQUEST, "This account does not exist.");
+    }
+
+    // Get all Collaboration Requests ever sent
+    Iterable<CollaborationRequest> allRequests = collaborationRequestRepository.findAll();
+
+    // Check whether the request was successful
+    if (allRequests == null) {
+      throw new GlobalException(HttpStatus.INTERNAL_SERVER_ERROR, "Could not retrieve requests.");
+    }
+
+    // Convert the iterable object to a list
+    List<CollaborationRequest> allRequestsList = new ArrayList<>();
+    allRequests.forEach(allRequestsList::add);
```

Comment:
This might be inefficient? If I understand correctly it iterates over the whole iterable and makes it a list?  Instead, in the collaborationRepository, you can define the return type you would like for the findAll method. Let me know if I can help with that, but something like List<CollaborationRequest> findAll(); 

Here is a resource
 https://www.amitph.com/spring-data-jpa-query-methods/